### PR TITLE
Fixed logic for downloading Fasta. 

### PR DESCRIPTION
**Pull request by ewels**
Also removed testing ENV var that specified nextflow version, so that tests run with latest version of Nextflow.

### CODE REVIEW

**Comment by ewels** [comment]
Ok, added a few more changes - the latest version of Nextflow gave lots of warnings about how we shouldn't be using `first()`. So I refactored how channels work for reference indexes a little. I also used the new `collect()` operator everywhere. This has a few benefits, the main one being that caching should be more reliable.

Finally, I added a new test script which does the same as the docker tests, but using `-profile devel` to work on UPPMAX. Running that test now.

File: main.nf [hunk 1]
```diff
@@ -96,7 +94,7 @@ if( params.gtf ){
     Channel
         .fromPath(params.gtf)
         .ifEmpty { exit 1, "GTF annotation file not found: ${params.gtf}" }
-        .toList()
+        .collect()
```
Comment (ewels) [inline main.nf]: Oops, I probably should have deleted this line.

File: tests/uppmax_test.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+data_path=$SNIC_NOBACKUP
+if [ -d "./test_data" ]
+then
+    data_path="./test_data"
+    echo "Found data directory in current working directory, using ./test_data/"
+fi
+
+curl --version >/dev/null 2>&1 || { echo >&2 "I require curl, but it's not installed. Aborting."; exit 1; }
+tar --version >/dev/null 2>&1 || { echo >&2 "I require tar, but it's not installed. Aborting."; exit 1; }
+nextflow -v >/dev/null 2>&1 || { echo >&2 "I require nextflow, but it's not installed. If you hava Java, run 'curl -fsSL get.nextflow.io | bash'. If not, install Java."; exit 1; }
```
Comment (Hammarn) [inline tests/uppmax_test.sh]: Helpful, I like it!

File: tests/uppmax_test.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+script_path="../main.nf"
+if [ -z $1]
+then
+    echo "No argument given, going to try to run ../main.nf"
+else
+    script_path=$1
+fi
```
Comment (ewels) [inline tests/uppmax_test.sh]: [Nope](http://tldp.org/HOWTO/Bash-Prog-Intro-HOWTO-6.html)

File: tests/uppmax_test.sh (hunk 2 above)
Comment (ewels) [inline tests/uppmax_test.sh]: This was written by @Galithil, I just copied it into this new script :)

**Pull request merged**